### PR TITLE
[feat]: [CDS-84438]: Support for multi select of events on events listing

### PR DESCRIPTION
**Pull request by adivishy1**
## Describe your changes
Add support for multi selection of events on the events listing page and added log lines. Have tested out the changes locally and its working as expected.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55338)

### CODE REVIEW

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55338)* status: 6 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/mapper/GitXWebhookMapper.java` line 206 at r2](https://reviewable.io/reviews/harness/harness-core/55338#-Nj79bhvAHdpJuoTZcFP:-Nj79bhvAHdpJuoTZcFQ:buf4yp4) ([raw file](https://github.com/harness/harness-core/blob/c959dfb6428f249ba353412e98d8c6fcc0c7be0c/136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/mapper/GitXWebhookMapper.java#L206)):*
> ```Java
> 
>   private EventStatusEnum getEventStatus(String eventStatus) {
>     if (EventStatusEnum.FAILED.value().equals(eventStatus)) {
> ```

can't we use switch case here?

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55338)* status: 6 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/service/GitXWebhookServiceImpl.java` line 112 at r2](https://reviewable.io/reviews/harness/harness-core/55338#-Nj79gMV9PjLkIu0nuJT:-Nj79gMV9PjLkIu0nuJU:b-9stviu) ([raw file](https://github.com/harness/harness-core/blob/c959dfb6428f249ba353412e98d8c6fcc0c7be0c/136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/service/GitXWebhookServiceImpl.java#L112)):*
> ```Java
>                 createGitXWebhookRequestDTO.getRepoName(), createGitXWebhookRequestDTO.getAccountIdentifier()),
>             USER_SRE, ex);
>       } catch (InternalServerErrorException exception) {
> ```

I don't see logic of having this separate catch block. What purpose does it serve?

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55338)* status: 6 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/service/GitXWebhookServiceImpl.java` line 148 at r2](https://reviewable.io/reviews/harness/harness-core/55338#-Nj79qJtAM7HLSkgme5i:-Nj79qJtAM7HLSkgme5j:b-l8ac1k) ([raw file](https://github.com/harness/harness-core/blob/c959dfb6428f249ba353412e98d8c6fcc0c7be0c/136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/service/GitXWebhookServiceImpl.java#L148)):*
> ```Java
>         return Optional.of(getGitXWebhookResponseList.get(0));
>       } catch (InternalServerErrorException exception) {
>         log.error("Unexpected error occurred while getting the GitX webhook {}",
> ```

what purpose does this exception block serve?

**Review by adivishy1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55338)* status: 6 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/mapper/GitXWebhookMapper.java` line 206 at r2](https://reviewable.io/reviews/harness/harness-core/55338#-Nj79bhvAHdpJuoTZcFP:-Nj7BgRv4gVxLlrfaMB9:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c959dfb6428f249ba353412e98d8c6fcc0c7be0c/136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/mapper/GitXWebhookMapper.java#L206)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

can't we use switch case here?
</blockquote></details>

Done.
___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/service/GitXWebhookServiceImpl.java` line 112 at r2](https://reviewable.io/reviews/harness/harness-core/55338#-Nj79gMV9PjLkIu0nuJT:-Nj7BmnBEFtg5RcYHQFD:blb48d8) ([raw file](https://github.com/harness/harness-core/blob/c959dfb6428f249ba353412e98d8c6fcc0c7be0c/136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/service/GitXWebhookServiceImpl.java#L112)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

I don't see logic of having this separate catch block. What purpose does it serve?
</blockquote></details>

agreed, removed it
___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/service/GitXWebhookServiceImpl.java` line 148 at r2](https://reviewable.io/reviews/harness/harness-core/55338#-Nj79qJtAM7HLSkgme5i:-Nj7BpeYDYxePoVA_doO:b-hfpv0l) ([raw file](https://github.com/harness/harness-core/blob/c959dfb6428f249ba353412e98d8c6fcc0c7be0c/136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/service/GitXWebhookServiceImpl.java#L148)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

what purpose does this exception block serve?
</blockquote></details>

removed

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 3 of 6 files at r1, 1 of 1 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55338)* status: 6 file(s) to review. 1 of 2 approvals obtained.

File: 136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/service/GitXWebhookServiceImpl.java [hunk 1]
```diff
@@ -130,14 +132,16 @@ public Optional<GetGitXWebhookResponseDTO> getGitXWebhook(GetGitXWebhookRequestD
         }
         List<GetGitXWebhookResponseDTO> getGitXWebhookResponseList = prepareGitXWebhooks(gitXWebhookList);
         if (getGitXWebhookResponseList.size() > 1) {
+          log.error(String.format(
+              "For the given key with accountIdentifier %s and gitXWebhookIdentifier %s found more than one unique record.",
```
Comment (deepakbolangady) [inline 136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/service/GitXWebhookServiceImpl.java:136]:
```suggestion
              "Found more than one unique record for the given key with accountIdentifier %s and gitXWebhookIdentifier %s.",
```
(Do we not want to write what component found more than one unique record? If so, incorporate this suggestion. Else, retain current structure and name the actor.)

File: 136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/service/GitXWebhookServiceImpl.java [hunk 2]
```diff
@@ -190,8 +194,11 @@ public UpdateGitXWebhookResponseDTO updateGitXWebhook(UpdateGitXWebhookCriteriaD
         }
         return UpdateGitXWebhookResponseDTO.builder().webhookIdentifier(updatedGitXWebhook.getIdentifier()).build();
       } catch (InternalServerErrorException exception) {
+        log.error("Unexpected error occurred while updating the GitX webhook {}",
```
Comment (deepakbolangady) [inline 136-git-sync-manager/src/main/java/io/harness/gitsync/gitxwebhooks/service/GitXWebhookServiceImpl.java:197]:
```suggestion
        log.error("Unexpected error occurred when updating the GitX webhook {}",
```

**Review by deepakbolangady — APPROVED** [review]
Approving with a couple of minor comments. Please incorporate if possible. Thanks.

**Comment by adivishy1** [comment]
trigger review

**Comment by adivishy1** [comment]
trigger messagecheck

**Pull request merged**